lucet-runtime/lucet-runtime-internals/src/instance.rs

```diff
@@ -1203,7 +1205,12 @@ impl Instance {
                 || self.state.is_yielded()
                 || self.state.is_bound_expired()
         );
-        self.state = State::Running { async_context };
+        
+        self.state = State::Running {
+            // SAFETY: We have to lie about the lifetime of async_context.
+            // However, State::Running will only last for as long as this function's lifespan.
+            async_context: unsafe { transmute(async_context) }
```

Comment:
I'm going to make the generic types here explicit, to prevent accidental sharp edges if the API evolves.